includes/admin/class-wc-rest-payments-orders-controller.php

```diff
@@ -124,7 +124,14 @@ public function capture_terminal_payment( $request ) {
 			}
 
 			// Do not process intents that can't be captured.
-			$intent = $this->api_client->get_intent( $intent_id );
+			$intent                   = $this->api_client->get_intent( $intent_id );
+			$intent_metadata          = is_array( $intent->get_metadata() ) ? $intent->get_metadata() : [];
+			$intent_meta_order_id_raw = $intent_metadata['order_id'] ?? '';
+			$intent_meta_order_id     = is_numeric( $intent_meta_order_id_raw ) ? intval( $intent_meta_order_id_raw ) : 0;
```

Comment:
As `$intent_metadata['order_id'] ?? ''` will tolerate non-array data structures, we can simplify this a little bit:

```php
$intent_meta_order_id = $intent->get_metadata()['order_id'] ?? '';
$intent_meta_order_id = is_numeric( $intent_meta_order_id ) ? (int) $intent_meta_order_id : 0;
```

Reply:
Fixed in 2297745.

Reply:
I had to revert this change because of the error mentioned in 18462d9.

Reply:
Gotcha, thanks for trying it out!